parachain/runtime/interlay/src/lib.rs

```diff
@@ -992,6 +992,73 @@ impl pallet_identity::Config for Runtime {
     type WeightInfo = ();
 }
 
+parameter_types! {
+    // One storage item; key size 32, value size 8; .
+    pub const ProxyDepositBase: Balance = deposit(1, 8);
+    // Additional storage item size of 33 bytes.
+    pub const ProxyDepositFactor: Balance = deposit(0, 33);
+    pub const MaxProxies: u16 = 32;
+    pub const MaxPending: u16 = 32;
+    pub const AnnouncementDepositBase: Balance = deposit(1, 8);
+    pub const AnnouncementDepositFactor: Balance = deposit(0, 66);
+}
+
+/// The type used to represent the kinds of proxying allowed.
+#[derive(
+    Copy,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    codec::Encode,
+    codec::Decode,
+    sp_runtime::RuntimeDebug,
+    codec::MaxEncodedLen,
+    scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+    Any,
+}
+
+impl Default for ProxyType {
+    fn default() -> Self {
+        Self::Any
+    }
+}
+
+impl InstanceFilter<RuntimeCall> for ProxyType {
+    fn filter(&self, c: &RuntimeCall) -> bool {
+        match self {
+            _ if matches!(c, RuntimeCall::Utility(..)) => true,
```

Comment:
The purpose of this isn't clear to me. It seems to have no use, can we remove?

Reply:
See this comment: https://github.com/AcalaNetwork/Acala/blob/a0251084b49b40d69262c84d13f4f9ed86729f1d/runtime/karura/src/lib.rs#L1291-L1293

Reply:
It is possible that this doesn't apply to `ProxyType::Any` but since in that case this is anyway a no-op, perhaps we should leave this so it is not forgotten in future updates?

Reply:
Okay I did some investigating, the way it works is if a `ProxyDefinition` exists it adds filters to the origin of the call: https://github.com/paritytech/substrate/blob/946507ba9ef13e263534176b7b74e26fc56efbd4/frame/proxy/src/lib.rs#L784

Say we dispatch `Utility::batch(Tokens::transfer(..))` (using `ProxyType::Any`) via the proxy pallet this will re-dispatch [here](https://github.com/paritytech/substrate/blob/946507ba9ef13e263534176b7b74e26fc56efbd4/frame/utility/src/lib.rs#L224) due to the above rule but then the inner call will be reevaluated using the other filter rules. So if we only support the `Any` permission I think we can drop the `Utility` filter.

Reply:
Ah ok, and then and calls inside a batch get dispatched with that additional filter. Ok that makes sense, thanks for looking into it